src/phasorpy/cursors.py

```diff
@@ -0,0 +1,299 @@
+"""Select regions of interest (cursors) from phasor coordinates.
+
+    The ``phasorpy.cursors`` module provides functions to:
+
+- create masks for regions of interests in the phasor space:
+
+  - :py:func:`mask_from_circular_cursor`
+
+- create masks for regions of interests in the polar space:
+
+  - :py:func:`mask_from_polar_cursor`
+
+- create a pseudo-color array of average signal from cursor regions:
+
+  - :py:func:`pseudo_color`
+
+"""
+
+from __future__ import annotations
+
+__all__ = [
+    'mask_from_circular_cursor',
+    'mask_from_polar_cursor',
+    'pseudo_color',
+]
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ._typing import (
+        Any,
+        ArrayLike,
+        NDArray,
+    )
+
+import numpy
+
+from phasorpy.color import CATEGORICAL
+
+
+def mask_from_circular_cursor(
+    real: ArrayLike,
+    imag: ArrayLike,
+    center_real: ArrayLike,
+    center_imag: ArrayLike,
+    /,
+    *,
+    radius: ArrayLike = 0.05,
+    axis: int = 0,
+) -> NDArray[numpy.bool_]:
+    """Return masks for circular cursors of phasor coordinates.
+
+    Parameters
+    ----------
+    real : array_like
+        Real component of phasor coordinates.
+    imag : array_like
+        Imaginary component of phasor coordinates.
+    center_real : array_like, shape (n,)
+        Real coordinates of circle centers.
+    center_imag : array_like, shape (n,)
+        Imaginary coordinates of circle centers.
+    radius : array_like, shape (n,)
+        Radii of circles.
+    axis : int, optional
+        Axis along which the masks are returned. Default is 0.
+
+    Returns
+    -------
+    masks : ndarray
+        Phasor coordinates masked for each circular cursor.
+
+    Raises
+    ------
+    ValueError
+        The array shapes of `real` and `imag`, or `center_real` and
+        `center_imag` do not match.
+        Any of the radii is negative.
+        The axis is out of bounds.
+
+    Examples
+    --------
+    Create mask for a single circular cursor:
+
+    >>> mask_from_circular_cursor([0.0, 0.0], [0.0, 0.5], 0.0, 0.5, radius=0.1)
+    array([False,  True])
+
+    Create masks for two circular cursors with different radius:
+
+    >>> mask_from_circular_cursor(
+    ...     [0.0, 1.0], [0.0, 0.5], [0.0, 1.0], [0.0, 0.4], radius=[0.1, 0.05]
+    ... )
+    array([[ True, False],
+           [False, False]])
+
+    """
+    real = numpy.atleast_1d(real)
+    imag = numpy.atleast_1d(imag)
+    center_real = numpy.asarray(center_real)
+    center_imag = numpy.asarray(center_imag)
+    radius = numpy.asarray(radius)
+
+    if real.shape != imag.shape:
+        raise ValueError(f'{real.shape=} != {imag.shape=}')
+    if center_real.shape != center_imag.shape:
+        raise ValueError(f'{center_real.shape=} != {center_imag.shape=}')
+    if numpy.any(radius < 0):
+        raise ValueError('all radii must be positive')
```

Comment:
Why? The code uses square of radius.

Reply:
You are right, this check is not needed.